socorro/lib/javautil.py

```diff
@@ -0,0 +1,113 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Utility functions for parsing and manipulating JavaStackTrace field
+contents.
+"""
+
+from more_itertools import peekable
+
+
+class JavaException(object):
+    def __init__(self, exception_class, exception_message, stack, additional):
+        self.exception_class = exception_class
+        self.exception_message = exception_message
+        self.stack = stack
+        self.additional = additional
+
+    def to_public_string(self):
+        """Builds stack trace with parts of the exception that don't contain PII
+
+        Currently, this skips the exception message which can contain data
+        related to the exception and might have PII in it.
+
+        It also skips the "Suppressed" and "Caused by" sections because those
+        can have exception messages, too.
+
+        """
+        text = '%s\n%s' % (
+            self.exception_class,
+            '\n'.join([
+                '\t%s' % line for line in self.stack
+            ])
+        )
+        return text
```

Comment:
This seems fine for now. It's possible that someone will have additional requirements for how this gets parsed and rendered. We can deal with those when that day comes.